pkg/controller/migration/convert/network.go

```diff
@@ -119,6 +130,107 @@ func handleNetwork(c *components, install *Installation) error {
 	return nil
 }
 
+func handleNonCalicoCNI(c *components, install *Installation) error {
+	plugin, err := getCNIPlugin(c)
+	if err != nil {
+		return err
+	}
+	if plugin == operatorv1.PluginCalico {
+		return nil
+	}
```

Comment:
You've implemented this correctly based on how I've put all he handlers in a slice. But I'm wondering aloud if handlers should include the logic to decide if they need to be run. I wonder if we should remove that slice and just have one big function that runs these kind of checks and calls the handlers accordingly.